[14.0][FIX] fastapi

xavier-bouquiaux opened the pull request:
fastapi_endpoint.py -> the url is relative so it does not work via the interface <br/>
~common.py -> odoo14 <> odoo16 (transaction vs cursor) for full details see sbi~

- [x] depends on https://github.com/OCA/server-tools/pull/2666

lmignon commented on fastapi/models/fastapi_endpoint.py:
```diff
@@ -80,10 +80,11 @@ def _check_root_path(self):
-            rec.docs_url = f"{rec.root_path}/docs"
-            rec.redoc_url = f"{rec.root_path}/redoc"
-            rec.openapi_url = f"{rec.root_path}/openapi.json"
+            rec.docs_url = f"{base_url}{rec.root_path}/docs"
+            rec.redoc_url = f"{base_url}{rec.root_path}/redoc"
+            rec.openapi_url = f"{base_url}{rec.root_path}/openapi.json"
```
Vetry strange since I tested the UI on V14 without these changes and everything is working.

lmignon replied:
Except that I added the base_url by hand ;-)

lmignon commented:
@xavier-bouquiaux Thank you for the PR. All the proposed changes make sense. Nevertheless, can you split the two commits plz? 
For the commit where you modify the odoo environement when running the tests, can you remove `for full details see sbi' from the commit message?

sbidoul committed:
[FIX] fastapi: use base_contextvars for thread mode compat

sbidoul committed:
[FIX] fastapi: fix docs urls

This is for 14.0 only.

sbidoul commented:
@lmignon @xavier-bouquiaux I updated this PR to move the contextvars patch to a `base_contextvars` module (https://github.com/OCA/server-tools/pull/2666).

sbidoul commented:
@xavier-bouquiaux my last change here should fix the issues you had when running with `workers=0`.

lmignon reviewed (approved):
Thank you @sbidoul for the fix. LGTM (Code review)

xavier-bouquiaux commented:
Thanks @sbidoul and @lmignon 

I've just rebased this on 14.0 as the depends has been merged

Let me know if it's ok or if i still need to do something

regards

xbo

lmignon commented:
@xavier-bouquiaux Can you confirm that this PR fix your issue with worker=0?

sbidoul committed:
[FIX] fastapi: patch http request to work around

Function declared as capable of handling request of type 'http' but called with a request of type

sbidoul committed:
[IMP] fastapi: Add POST demo reouter

sbidoul committed:
[IMP] fastapi: disable csrf on fastapi routes

sbidoul commented:
/ocabot merge patch
